windows/src/desktop/kmshell/main/Keyman.System.UpdateStateMachine.pas

```diff
@@ -558,14 +563,27 @@ function TUpdateStateMachine.CurrentStateName: string;
   Result := CurrentState.ClassName;
 end;
 
-function TUpdateStateMachine.ReadyToInstall: Boolean;
+function TUpdateStateMachine.ValidateReadyToInstall: Boolean;
 begin
   if not IsCurrentStateAssigned then
     Exit(False);
-  if (CurrentState is WaitingRestartState) and not HasKeymanRun then
-    Result := True
-  else
-    Result := False;
+
+  if not (CurrentState is WaitingRestartState) then
+    Exit(False);
+
+  // Verify conditions since entering WaitingRestartState:
+  // If the cache is not valid or automatic updates have
+  // been disabled clear the cache and return to the idle state.
+  if not (TUpdateCheckStorage.CheckMetaDataForUpdate and Self.GetAutomaticUpdates) then
```

Comment:
```suggestion
  if not TUpdateCheckStorage.CheckMetaDataForUpdate or not Self.GetAutomaticUpdates then
```

Unless it is important that both functions are called, then it's clearer like this, and matches the comment, I think? (De Morgan's law)

If it is important that both functions are called, then that probably should be called out as a side-effect.